java/maven/src/org/netbeans/modules/maven/options/MavenVersionSettings.java

```diff
@@ -1,76 +0,0 @@
-/*
- * Licensed to the Apache Software Foundation (ASF) under one
- * or more contributor license agreements.  See the NOTICE file
- * distributed with this work for additional information
- * regarding copyright ownership.  The ASF licenses this file
- * to you under the Apache License, Version 2.0 (the
- * "License"); you may not use this file except in compliance
- * with the License.  You may obtain a copy of the License at
- *
- *   http://www.apache.org/licenses/LICENSE-2.0
- *
- * Unless required by applicable law or agreed to in writing,
- * software distributed under the License is distributed on an
- * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
- * KIND, either express or implied.  See the License for the
- * specific language governing permissions and limitations
- * under the License.
- */
-package org.netbeans.modules.maven.options;
-
-import java.util.prefs.Preferences;
-import org.openide.util.NbPreferences;
-
-/**
- * Preferences class for externalizing the hardwired plugin versions to
- * allow changes by advanced users?
- * @author mkleint
- */
-public final class MavenVersionSettings {
```

Comment:
the class is about artifact versions, not the maven version. Analog to [RepositoryPreferences](https://github.com/apache/netbeans/blob/master/java/maven.indexer/src/org/netbeans/modules/maven/indexer/api/RepositoryPreferences.java) which persists the repo config to Preferences, this should carry a similar name in my opinion. 

Based on comments of the half finished implementation it seemed to follow similar design.

Since it is a final friend API, carried barely any functionality was and not used anywhere it should be ok to change.

Reply:
renamed it back and deprecated a few things instead